src/cascadia/TerminalSettingsModel/ActionArgs.h

```diff
@@ -89,6 +92,12 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
             JsonUtils::GetValueForKey(json, TabTitleKey, args->_TabTitle);
             JsonUtils::GetValueForKey(json, ProfileIndexKey, args->_ProfileIndex);
             JsonUtils::GetValueForKey(json, ProfileKey, args->_Profile);
+
+            if (const auto tabColor{ JsonUtils::GetValueForKey<std::optional<til::color>>(json, TabColorKey) })

```

Comment:
I think if you make TabColor an `Windows::Foundation::IReference<Windows::UI::Color>`, like `Profile::Foreground`, you should be able to just do 
```c++
    JsonUtils::GetValueForKey(json, TabColorKey, _TabColor );
```

Reply:
@zadjii-msft - actually I started with TabColor being a Windows::UI::Color, but then I saw that SetTabColor command uses til:color, thus I switched to it in the sake of code consistency. I suggest to either leave it as is or I can fix the SetTabColor command to use Windows::UI::Color. WDYT?

Reply:
Ah heck, that should probably be a `Windows::UI::Color` too. Sorry to bounce you around like this!

Reply:
@zadjii-msft - NP. I am fixing this!